### PR TITLE
Document qsimcirq GPU usage

### PR DESCRIPTION
**Pull request by 95-martin-orion**
Follow-up to #398, providing guidance on how to use GPUs from qsimcirq.

### CODE REVIEW

File: docs/cirq_interface.md [hunk 1]
```diff
@@ -148,7 +148,26 @@ matrices, if one is specified.
...
+may improve performance. In order to use this feature, qsim must be compiled on
+a device with the [CUDA toolkit](https://developer.nvidia.com/cuda-downloads)
+and run on a device with available NVIDIA GPUs.
+
+Compilation for GPU follows the same steps outlined in the
+[Compiling qsimcirq](./cirq_interface.md#compiling-qsimcirq) section.
+
+`QSimOptions` provides four parameters to configure GPU execution:
+* use_gpu: whether to use GPU instead of CPU for simulation.
+* gpu_sim_threads: number of threads per CUDA block to use for the GPU
```
Comment (sergeisakov) [inline docs/cirq_interface.md:168]: Could you clarify that the last three parameters are for performance fine-tuning and the user does not need to specify them as the default values are good enough in most cases?

File: docs/cirq_interface.md [hunk 2]
```diff
@@ -76,11 +76,11 @@ This circuit can then be simulated using either `QSimSimulator` or
...
 average PC, or up to 40 qubits on high-performance VMs).
 Options for the simulator, including number of threads and verbosity, can be
-set with the `qsim_options` field using the `qsim_base` flag format defined in
-the [usage docs](./usage.md).
+set with the `qsim_options` field, which accepts a `QSimOptions` object as
+defined in [qsim_simulator.py](https://github.com/quantumlib/qsim/blob/master/qsimcirq/qsim_simulator.py).
 
 ```
-qsim_options = {'t': 8, 'v': 0}
+qsim_options = qsimcirq.QSimOptions(cpu_threads=8, verbosity=0)
```
Comment (sergeisakov) [inline docs/cirq_interface.md:83]: The old format is still available, right?

**Review by sergeisakov — CHANGES_REQUESTED** [review]

File: docs/cirq_interface.md (hunk 1 above)
Comment (95-martin-orion) [reply docs/cirq_interface.md]: Done.

File: docs/cirq_interface.md (hunk 2 above)
Comment (95-martin-orion) [reply docs/cirq_interface.md:88]: Added some clarification and an equivalent dict format to the code block.

**Review by sergeisakov — APPROVED** [review]
LGTM

**Pull request merged**